iset.mm

```diff
@@ -125615,6 +125689,168 @@ reduced fraction representation (no common factors, denominator
 $)
 
 
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Topology
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Topological spaces
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+  A topology on a set is a set of subsets of that set, called open sets, which
+  satisfy certain conditions.  One condition is that the whole set be an open
+  set.  Therefore, a set is recoverable from a topology on it (as its union),
+  and it may sometimes be more convenient to consider topologies without
+  reference to the underlying set.
+
+$)
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Topologies
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $( Token for the class of topologies. $)
+  $c Top $.
+
+  $( Syntax for the class of topologies. $)
+  ctop $a class Top $.
+
+  ${
+    $d x y z $.
+    $( Define the class of topologies.  It is a proper class.  See ~ istopg and
+       ~ istopfin for the corresponding characterizations, using respectively
+       binary intersections like in this definition and nonempty finite
+       intersections.
+
+       The final form of the definition is due to Bourbaki (Def. 1 of
+       [BourbakiTop1] p.  I.1), while the idea of defining a topology in terms
+       of its open sets is due to Aleksandrov.  For the convoluted history of
+       the definitions of these notions, see
+
+       Gregory H. Moore, _The emergence of open sets, closed sets, and limit
+       points in analysis and topology_, Historia Mathematica 35 (2008)
+       220--241.
+
+       (Contributed by NM, 3-Mar-2006.)  (Revised by BJ, 20-Oct-2018.) $)
+    df-top $a |- Top = { x | ( A. y e. ~P x U. y e. x /\
+                             A. y e. x A. z e. x ( y i^i z ) e. x ) } $.
```

Comment:
Should the definition be changed to
```
  df-top $a |- Top = { x | ( A. y e. ~P x U. y e. x /\ A. y e. x A. z e. x ( -. y = z -> ( y i^i z ) e. x ) ) } $.
```
I haven't thought nor read anything about that.  Has someone seen intuitionistic treatments of topology ? @digama0 ?

Reply:
> Should the definition be changed to
> 
> ```
>   df-top $a |- Top = { x | ( A. y e. ~P x U. y e. x /\
>                        A. y e. x A. z e. x ( -. y = z -> ( y i^i z ) e. x ) ) } $.
> ```
> 
> I haven't thought nor read anything about that. Has someone seen intuitionistic treatments of topology ? @digama0 ?

I looked a bit at Franka Waaldijk, modern intuitionistic topology, 1996, cited at the bottom of https://ncatlab.org/nlab/show/topology , but if it had answers to this sort of question I didn't find them. It seemed to be about more advanced topics.

My hunch is that we don't need to change the definition but I suspect it will become clear one way or the other as we try to prove more theorems. To be honest I'm still a bit lost in all the topology related definitions and notations in set.mm and I don't really know how many of them I need for my monotonic intermediate value theorem which is my focus at the moment.

Reply:
Thanks for the pointer, I'll have a look. I would say: maybe do not go too fast before we are settled on the definition, since this may cause some rework. Indeed, I tend to believe that the definition I proposed is more in line with intuitionism (though I really haven't thought enough about it).

Reply:
On page 50 of http://www.fwaaldijk.nl/modern%20intuitionistic%20topology.pdf the intersection axiom (O_3) is stated as in classical maths, but attention is restricted to topological spaces satisfying an axiom (O_4) which has to do with decidability of apartness.

Reply:
I had a look at Bauer's "Five stages of accepting constructive mathematics". So between the two proposals above, I now think it is better to keep the classical one. Indeed, his definition (when converting from locales) is the same as in the classical case.

By the way, it's funny that you could prove stability under finite intersections which are nonempty, and not only inhabited. I'll have a look at the proof.

Is it interesting to see which results remain true if we adopt the more lax definition of a topology ?

Bauer defines topological spaces using locales. Maybe it would be better to do that in iset.mm ? (this can be done later, though)

Reply:
> By the way, it's funny that you could prove stability under finite intersections which are nonempty, and not only inhabited. I'll have a look at the proof.

If you know the set is finite the distinction between nonempty and inhabited goes away by https://us.metamath.org/ileuni/fin0or.html